ElectronClient/gui/NoteEditor/NoteBody/CodeMirror/utils/useExternalPlugins.ts

```diff
@@ -0,0 +1,15 @@
+import { PluginStates } from 'lib/services/plugins/reducer';
+import contentScriptsToCodeMirror from 'lib/services/plugins/utils/contentScriptsToCodeMirror';
+
+
+export default function useExternalPlugins(CodeMirror: any, plugins: PluginStates) {
+	const contentScripts = contentScriptsToCodeMirror(plugins);
+	for (const contentScript of contentScripts) {
+		for (const addon of contentScript.module.addons) {
+			require(`codemirror/addon/${addon}`);
+		}
+		contentScript.module.plugin(CodeMirror);
+	}
+	// TODO: This should only be run on started, currently it is run on every key stroke
+	console.log('Loading CodeMirror Plugins');
```

Comment:
I guess just wrapping it in useEffect would do the job. The "plugins" state variable is stable and won't change often.

Reply:
Good call, nice and simple.